### DISCUSSION TITLE
How rsbuild support `webpack` multicompiler

### DISCUSSION THREAD

**Discussion by tianyingchun** [web-infra-dev/rsbuild#2246] [General]
how to make`resbuld` to support `multi compiler` of webpack

https://webpack.js.org/api/node/#multicompiler

**Comment by chenjiahan** [comment]
Rsbuild will internally create multiCompiler when the [output.targets](https://rsbuild.dev/config/output/targets) contains more than one value.

For example, if `output.targets` is `['web', 'node']`, Rsbuild will create one compiler for the web target and another compiler for the node target.

And currently this is the only way to create a multiCompiler in Rsbuild.

**Reply by tianyingchun** [reply]
For example, I think this is useful for the same target web Turn on multiple compiler debugging modes while launching custom entries

**Comment by tianyingchun** [comment]
We have plans to add similar webpack multi compiler APi Allow us to manually start multiple independent processes at the same time?

**Reply by chenjiahan** [reply]
You can use Rsbuild's JavaScript API to create multiple rsbuild instances and call their build methods at the same time.
